src/lib/structures/expr.mli

```diff
@@ -309,17 +309,26 @@ val make_triggers:
 (** [make_triggers f binders decl menv] generate multi-triggers for the
     formula [f] and binders [binders].
 
-    For axioms, we try to produce multi-triggers in the following order
-    (if we succeed to produce at least one multi-trigger, we don't try other
-    strategies):
-    1. In greedy mode:
-    - Mono-triggers without variables;
-    - Mono-triggers with variables;
-    - Multi-triggers without variables;
-    - Multi-triggers with variables.
-      2. Otherwise:
-    - Mono and multi-triggers without variables;
-    - Mono and multi-triggers with variables.
+    The following documentation describes the strategies used by SatML
+    to generate multi-triggers.
+
+    Assuming that [Options.get_triggers_var ()] is [true] (the default),
+    we do not allow trigger variables in frugal, normal and greedy mode
+    but we allow them in greedier mode.
+
+    While generating multi-triggers (with at least two patterns), we can
+    escape free variables of their patterns that aren't bound in the quantified
+    formula.
```

Comment:
I cannot parse this sentence. Maybe it would be more clear with a definition of "escaping variable" and an example?

Reply:
I tried to reformulate and add an example. If I understand now what is the purpose of these substitutions. 

Honestly, I don't understand the purpose of this feature. Consider the example:
```
logic P: int -> prop
logic Q: int, int -> prop
logic a: int

axiom a: forall z: int. forall x:int. P(x) -> forall y:int. Q(y, z)
goal g1: P(1) -> Q(a, 1)
```
`Alt-Ergo` will generate the multi-trigger `{P(x), Q(_,z)}`

As I tried to understand the purpose of this feature, I documented more functions in `Triggers`, sorry :/

Reply:
`y` is not a quantified variable of the toplevel quantifier. As I understand it, the multi-trigger `{P(x), Q(_, z)}` will instantiate the axiom when there is a term of the form `P(x)` (binding `x`) and of the form `Q(_, z)` (binding `z`) in the context to generate `P(x) -> forall y:int. Q(y, z)`. Later we might instanciate the inner `forall` quantifier but only if `P(x)` holds.

Reply:
Sure, I understood that but I don't understand why in `greedy` mode we first try multi-trigger with escaped variables. I expected the reverse order.

Reply:
What I don't understand is why we try both — from what I understand, the only difference is whether we bind `z` or not, so doing only the version with escaped variables looks sufficient.

Reply:
I will check if we can remove the other one in the separated PR.